modules/google-speech/src/backend/middleware.ts

```diff
@@ -0,0 +1,149 @@
+import axios from 'axios'
+import * as sdk from 'botpress/sdk'
+import FormData from 'form-data'
+import LRUCache from 'lru-cache'
+import ms from 'ms'
+import { v4 as uuidv4 } from 'uuid'
+
+import { GoogleSpeechClient } from './client'
+import { Clients } from './typings'
+
+const INCOMING_MIDDLEWARE_NAME = 'googleSpeech.speechToText'
+const OUTGOING_MIDDLEWARE_NAME = 'googleSpeech.textToSpeech'
+
+export class Middleware {
+  private readonly timeout = '15s'
+  private isTextToSpeechCache: LRUCache<string, boolean>
+
+  constructor(private bp: typeof sdk, private clients: Clients) {
+    this.isTextToSpeechCache = new LRUCache({ maxAge: ms('1m') })
+  }
+
+  public setup() {
+    this.bp.events.registerMiddleware({
+      description: 'Converts audio content to text using google speech-to-text.',
+      direction: 'incoming',
+      handler: this.incomingHandler.bind(this),
+      name: INCOMING_MIDDLEWARE_NAME,
+      order: -1,
+      timeout: this.timeout
+    })
+
+    this.bp.events.registerMiddleware({
+      description: 'Converts text to audio using google text-to-speech.',
+      direction: 'outgoing',
+      handler: this.outgoingHandler.bind(this),
+      name: OUTGOING_MIDDLEWARE_NAME,
+      order: -1,
+      timeout: this.timeout
+    })
+  }
+
+  private async incomingHandler(event: sdk.IO.IncomingEvent, next: sdk.IO.MiddlewareNextCallback) {
+    // TODO: why not just use the audio content type?
+    if (event.payload.type !== 'voice') {
+      return next(undefined, false, true)
+    }
```

Comment:
I remember there was an explanation for this, but I don't remember what was the reason we decided to use a "voice" payload instead of an "audio" payload.